src/parser_TEST.cc

```diff
@@ -767,6 +767,52 @@ TEST(Parser, ElementFilePath)
   EXPECT_EQ("", directModel->FilePath());
 }
 
+/////////////////////////////////////////////////
+/// Check for missing element segfault, #590
+TEST(Parser, MissingRequiredElement)
+{
+  const std::string testString = R"(<?xml version="1.0" ?>
+    <sdf version="1.8">
+      <world name="default">
+        <model name="test_model">
+          <link name="link_0"/>
+          <link name="link_1"/>
+          <joint name="joint" type="revolute">
+            <parent>link_0</parent>
+            <axis>
+              <xyz>1 0 0</xyz>
+              <limit>
+                <lower>-1</lower>
+                <upper>1</upper>
+              </limit>
+            </axis>
+          </joint>
+        </model>
+      </world>
+    </sdf>)";
+
+  auto findFileCb = [](const std::string &_uri)
+  {
+    return sdf::testing::TestFile("integration", "model", _uri);
+  };
+
+  sdf::ParserConfig config;
+  config.SetFindCallback(findFileCb);
```

Comment:
nit: Since there are no `//include` elements, is the `findFileCb` needed?

Reply:
Thanks for catching that, removed in https://github.com/ignitionrobotics/sdformat/pull/610/commits/1c79f2fd80f84b801ce5d490ed9f764379805096